java/shopping-cart/src/main/statefun-config-sample/module.yaml

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+version: "3.0"
+module:
+  meta:
+    type: remote
+  spec:
+    endpoints:
+      - endpoint:
+          meta:
+            kind: http
+          spec:
+            typename:
+              namespace: example
+              type: greeter
+            urlPathTemplate: http://shopping-cart-service.statefun.svc:8080/statefun
+            maxNumBatchRequests: 500
+            timeouts:
+              call: 2min
+    ingresses:
+      - ingress:
+          meta:
+            type: statefun.kafka.io/universal-ingress
+            id: example.com/shopping-cart-updates
+          spec:
+            address: cp-cp-kafka.confluent-platform.svc:9092
+            consumerGroupId: my-group-id
+            topics:
+              - topic: shopping-cart-updates
+                # ? Unclear which type to use if multiple cart-related messages
```

Comment:
🤔 This is a very good question!
Yes, I think that you are right, we don't have meta types.
One way to approach this is indeed adding a `kind` field in your JSON schema, and to handle this with Jackson.
The typeUrl that you can write here will be a bit generic, for example `com.example/ShoppingCartEvents`.
And delegate the polymorphic deserialization to Jackson.

Another way, specifically for the cart function is:
1. ignoring the typeUrl and obtaining the raw bytes (`message.getRawValue()`).
2. using Jackson's readTree 
3. inspect the real type 
4. convert the tree to the specific object.

We need to see what is easier to grok in that case.

Reply:
As this is not a "runnable" example, I propose then to leave it as is for the scope of this PR and address it as a follow-up.